### DISCUSSION TITLE
Ping-Pong Functionality in quarkus

### DISCUSSION THREAD

**Discussion by adityakrk** [quarkusio/quarkus#41228] [Q&A]
Hi ,
I have implemented ping-pong functionality for jakarta.websocket using quarkus . so i wrote some TestClient to check the functionality .
Firstly i have connected to the server using TestClient once the connection is established onOpen method initiate the heartbeat to sendPing to the client for every 30sec. 

```
@OnMessage
   public void onPongMessage(Session session, PongMessage pongMessage) {
}
```
But i want client shouldn't respond with automatic pongmessage if server sends ping .Is there any attribute to stop automatic pong . Please find the details of the version i have used and suggest me the version which work as excepted .
Quarkus Version : 3.4.1
io.quarkus:quarkus-websockets :3.4.1

**Comment by maxandersen** [comment]
Can you provide reproducer ?

**Reply by geoand** [reply]
Also `3.4.1` is rather old. Best to try with the latest version of Quarkus

**Reply by nithinbandaru1** [reply]
@adityakrk

I suspect that actually here client is automatically responding to server pings that are sent by this Heartbeat class.
So, server session handler method would be receiving the response of server ping that is client pong.

Clients OnPongMessage handler will be hit when server sends a pong message for client's ping message.

One thing I am not sure is how is if Quarkus WebSocket client responds to server pings automatically with pong.
**This is interesting to know. I Just wonder if there is a way to disable auto responding functionality somehow.**

I think there is some confusion what is ping message and pong message from your side. Client Session OnPongMessage is actually a response pong message that is received from server when client sends ping.

@maxandersen @geoand  Please correct me if I am wrong here.
